nats/handler.go

```diff
@@ -0,0 +1,197 @@
+package nats
+
+import (
+	"os"
+	"sync"
+
+	"github.com/nats-io/nats.go"
+)
+
+// EventHandler handles subscribing and queue subscribing on nats and jetstream
+type EventHandler interface {
+	SubscribeToQueueUsingChannel(subject string, group string, msgChannel chan *nats.Msg) error
+	SubscribeToQueue(subject string, group string, cb nats.MsgHandler) error
+	Subscribe(subject string, cb nats.MsgHandler) error
+}
+
+type eventHandlerBase struct {
+	subs        []*nats.Subscription
+	mutex       sync.Mutex
+	termination <-chan os.Signal
+}
+
+// handlerConfig config for nats and jetstream
+type handlerConfig struct {
+	autoUnsubscribeOnShutdown bool
+}
+
+// NatsEventHandler nats event handler structure which implements EventHandler interface
+type NatsEventHandler struct {
+	eventHandlerBase
+	nc     *nats.Conn
+	config *handlerConfig
+}
+
+// JsEventHandler jetstream event handler structure which implements EventHandler interface
+type JsEventHandler struct {
+	eventHandlerBase
+	js     nats.JetStreamContext
+	config *handlerConfig
+}
+
+// For Checking compatibility
+var (
+	_ EventHandler = (*NatsEventHandler)(nil)
```

Comment:
I think I prefer to put those lines into the test file, even in a test so it's more clear this is testing the interface compliance. It can eventually save few bytes of  memory footprint.